### PR TITLE
Improve build system + prepare for 0.22

### PR DESCRIPTION
**Pull request by benesch**
Probably easiest just to look at the commit messages and the updates to the changelog/README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,33 +114,38 @@ Add this to your `Cargo.toml`:
...
+will be used to compile and statically link the library. The `dynamic-linking`
+feature can be used to instead dynamically link rdkafka to the system's version
+of librdkafka. Example:
 
 ```toml
-[dependencies.rdkafka]
-version = "~0.21"
-features = ["ssl", "sasl"]
+[dependencies]
+rdkafka = { version = "0.22", features = ["cmake-build", "dynamic-linking"] }
```
Comment (fede1024) [inline README.md]: Is cmake-build necessary when using dynamic linking?

Comment (benesch) [reply README.md]: D'oh, good catch, thanks!

**Comment by benesch** [comment]
Thank you for building the library in the first place! Happy to keep things moving forward.

**Comment by fede1024** [comment]
Should I build a new release? Or are there still open issues that we want to solve first?

**Comment by benesch** [comment]
Yes, please! Folks seem to be ready for a new release, and there’s nothing else that seems urgent. (I want to get to the futures stuff ASAP, but better to put that in 0.23 instead of the next release, since there’s a lot of stuff in the ecosystem that’s still on tokio 0.1, and those folks could benefit from the past six months of compatible improvements here.)

**Pull request merged**